src/export/StructureDefinitionExporter.ts

```diff
@@ -482,6 +482,20 @@ export class StructureDefinitionExporter implements Fishable {
           // Note: newElement() method automatically adds the new element to its structDef.elements
           const newElement = structDef.newElement(rule.path);
           newElement.applyAddElementRule(rule, this);
+          const lastIdPart = newElement.id.slice(newElement.id.lastIndexOf('.') + 1);
+          // see http://hl7.org/fhir/elementdefinition-definitions.html#ElementDefinition eld-19 and eld-20
+          const nameErrors: string[] = [];
+          if (!/^[^\s\\.,:;\\'"\/|?!@#$%&*()\[\]{}]+(\[x\])?$/.test(lastIdPart)) {
+            nameErrors.push('cannot include some special characters');
+          }
+          if (lastIdPart.length > 64) {
+            nameErrors.push('must be at most 64 characters long');
+          }
+          if (nameErrors.length > 0) {
+            logger.error(`Element names ${nameErrors.join(' and ')}.`, rule.sourceInfo);
+          } else if (!/^[A-Za-z][A-Za-z0-9]*$/.test(lastIdPart)) {
```

Comment:
Won't this warn on valid choice paths like `Foo.value[x]`? We shouldn't warn on those.